game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -60,10 +60,16 @@ public void load(final @Nullable GameData data, final @Nullable String fileName)
 
   public void load(final GameChooserEntry entry) {
     fileName = null;
-    setGameData(entry.getGameData());
-    ClientSetting.defaultGameName.setValue(entry.getGameData().getGameName());
-    ClientSetting.defaultGameUri.setValue(entry.getUri().toString());
-    ClientSetting.flush();
+    if (entry == null) {
+      ClientSetting.defaultGameName.resetValue();
+      ClientSetting.defaultGameUri.resetValue();
+      ClientSetting.flush();
+    } else {
+      setGameData(entry.getGameData());
+      ClientSetting.defaultGameName.setValue(entry.getGameData().getGameName());
+      ClientSetting.defaultGameUri.setValue(entry.getUri().toString());
+      ClientSetting.flush();
+    }
```

Comment:
Actually I'd place `ClientSetting.flush()` outside of the individual if cases

Reply:
Agree, it's better to keep the conditional branches focused to only the things that are different.

Reply:
Updated: c212dfdd06b6c97f353ffccf036875ffcde606b9